Userland/Libraries/LibGfx/ImageFormats/ImageDecoder.h

```diff
@@ -28,6 +30,15 @@ struct VectorImageFrameDescriptor {
     int duration { 0 };
 };
 
+struct BasicMetadata {
```

Comment:
wdyt about making this a pure virtual interface and having some class that derives from it and that contains a ref to ExifMetadata internally (BasicExifMetadata or what)? Other image formats could then also implement this interface.

How did you come up with manufacturer, model, software, creation_time, author as the entries here?

The concrete metadata formats that matter are probably exif, iptc, xmp.

PNG has this https://www.w3.org/TR/2003/REC-PNG-20031110/#11keywords

TIFF probably has something like that list somewhere too.

Reply:
> How did you come up with manufacturer, model, software, creation_time, author as the entries here?

I tried to have something fairly compatible with most format, but that's probably not doable. So I tried something else: I completely changed the design to be way more flexible. This is done by having the basic metadata already being passed as an iterable. So every format can provide their full tag list and the important subset for the property window.